packages/vite-node/src/cli.ts

```diff
@@ -75,8 +75,15 @@ async function run(files: string[], options: CliOptions = {}) {
   // provide the vite define variable in this context
   await runner.executeId('/@vite/env')
 
-  for (const file of files)
-    await runner.executeFile(file)
+  for (const file of files) {
+    try {
+      await runner.executeFile(file)
+    }
+    catch (e) {
+      if (!options.watch)
+        throw e
+    }
```

Comment:
I think we should still print out the error and set exitcode

Reply:
Sure any need to set the exit code back to 0 on hmr?